addons/base/tests/views.py

```diff
@@ -30,7 +30,7 @@ def test_oauth_start(self):
         redirect_params = urlparse.parse_qs(redirect_url.query)
         provider_url = urlparse.urlparse(self.Provider().auth_url)
         provider_params = urlparse.parse_qs(provider_url.query)
-        for param, value in redirect_params.items():
+        for param, value in list(redirect_params.items()):
```

Comment:
As expected, `d.items()` -> `list(d.items()`, but it's unnecessary in this context.

@Johnetordoff Can you see if you can fix the places where it's unnecessary to coerce to a list?